opm/simulators/linalg/PressureTransferPolicy.hpp

```diff
@@ -0,0 +1,158 @@
+        coarseLevelMatrix_.reset(new CoarseMatrix(fineLevelMatrix.N(), fineLevelMatrix.M(), CoarseMatrix::row_wise));
+        auto createIter = coarseLevelMatrix_->createbegin();
+
+        for (const auto& row : fineLevelMatrix) {
+            for (auto col = row.begin(), cend = row.end(); col != cend; ++col) {
+                createIter.insert(col.index());
+            }
+            ++createIter;
+        }
+
+        calculateCoarseEntries(fineOperator);
+        coarseLevelCommunication_.reset(communication_, [](Communication*) {});
+
+        this->lhs_.resize(this->coarseLevelMatrix_->M());
+        this->rhs_.resize(this->coarseLevelMatrix_->N());
+        using OperatorArgs = typename Dune::Amg::ConstructionTraits<CoarseOperator>::Arguments;
+        OperatorArgs oargs(*coarseLevelMatrix_, *coarseLevelCommunication_);
+        this->operator_.reset(Dune::Amg::ConstructionTraits<CoarseOperator>::construct(oargs));
+    }
+
+    virtual void calculateCoarseEntries(const FineOperator& fineOperator) override
+    {
+        const auto& fineMatrix = fineOperator.getmat();
+        *coarseLevelMatrix_ = 0;
+        auto rowCoarse = coarseLevelMatrix_->begin();
+        for (auto row = fineMatrix.begin(), rowEnd = fineMatrix.end(); row != rowEnd; ++row, ++rowCoarse) {
+            assert(row.index() == rowCoarse.index());
+            auto entryCoarse = rowCoarse->begin();
+            for (auto entry = row->begin(), entryEnd = row->end(); entry != entryEnd; ++entry, ++entryCoarse) {
+                assert(entry.index() == entryCoarse.index());
+                double matrix_el = 0;
+                if (transpose) {
+                    auto bw = weights_[entry.index()];
+                    for (size_t i = 0; i < bw.size(); ++i) {
+                        matrix_el += (*entry)[pressure_var_index_][i] * bw[i];
+                    }
+                } else {
+                    auto bw = weights_[row.index()];
+                    for (size_t i = 0; i < bw.size(); ++i) {
+                        matrix_el += (*entry)[i][pressure_var_index_] * bw[i];
+                    }
+                }
+                (*entryCoarse) = matrix_el;
+            }
+        }
+        assert(rowCoarse == coarseLevelMatrix_->end());
+    }
+
+    virtual void moveToCoarseLevel(const typename ParentType::FineRangeType& fine) override
+    {
+        // Set coarse vector to zero
+        this->rhs_ = 0;
+
+        auto end = fine.end(), begin = fine.begin();
+
+        for (auto block = begin; block != end; ++block) {
+            auto bw = weights_[block.index()];
+            double rhs_el = 0.0;
+            if (transpose) {
+                rhs_el = (*block)[pressure_var_index_];
+            } else {
+                for (size_t i = 0; i < block->size(); ++i) {
+                    rhs_el += (*block)[i] * bw[i];
+                }
+            }
+            this->rhs_[block - begin] = rhs_el;
+        }
+
+        this->lhs_ = 0;
```

Comment:
Communication missing in the parallel case?

Reply:
There is no communication in the BlackoilAmg version either. If `fine` is ok then the coarse vector should also be? We think that it should have the same "quality" (valid, unique, additive or consistent) as the fine vector.

Reply:
If this must be modified, should also moveToFineLevel() be modified?